src/OrleansLoggerProvider/LegacyOrleansLogger.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions.Internal;
+using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+
+namespace Orleans.Extensions.Logging.Legacy
+{
+    /// <summary>
+    /// LegacyOrleansLogger supports legacy orleans logging features, including <see cref="ILogConsumer"/>, <see cref="ICloseableLogConsumer">,
+    /// <see cref="IFlushableLogConsumer">, <see cref="Severity">. 
+    /// </summary>
+    [Obsolete("Use Microsoft.Extensions.Logging built-in loggers")]
+    public class LegacyOrleansLogger : ILogger
+    {
+        private readonly TimeSpan flushInterval = Debugger.IsAttached ? TimeSpan.FromMilliseconds(10) : TimeSpan.FromSeconds(1);
+        private DateTime lastFlush = DateTime.UtcNow;
+
+        private readonly IList<ILogConsumer> logConsumers;
+        private readonly string name;
+        private readonly IList<IFlushableLogConsumer> flushableConsumers;
+        private readonly LoggerType loggerType;
+        private readonly IPEndPoint ipEndPoint;
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="categoryName"></param>
+        /// <param name="logConsumers"></param>
+        /// <param name="maxSeverityLevel"></param>
+        public LegacyOrleansLogger(string categoryName, IList<ILogConsumer> logConsumers, IPEndPoint ipEndPoint)
+        {
+            this.logConsumers = logConsumers;
+            this.flushableConsumers = logConsumers.OfType<IFlushableLogConsumer>().ToList();
+            this.name = categoryName;
+            this.loggerType = DetermineLoggerType(categoryName);
+            this.ipEndPoint = ipEndPoint;
+        }
+
+        /// <inheritdoc/>
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            if (state == null)
+            {
+                throw new ArgumentNullException(nameof(state));
+            }
+            //TODO: support logging scope
+            return NullScope.Instance;
+        }
+
+        /// <inheritdoc/>
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
```

Comment:
Why no validating of log levels here?

Reply:
no need, it is taken care by log filters